src/BlockingQueue.h

```diff
@@ -0,0 +1,53 @@
+#ifndef BLOCKINGQUEUE_H
+#define BLOCKINGQUEUE_H
+#pragma once
+
+#include <mutex>
+#include <condition_variable>
+#include <deque>
+
+template <typename T, typename  Container = std::deque<T>>
+class BlockingQueue
+{
+private:
+	std::mutex d_mutex;
+	std::condition_variable d_condition;
+	Container d_queue;
```

Comment:
This class was copied and pasted code from stack overflow. I'll get it now inline with the project's style.